scripts/aws_replicate.py

```diff
@@ -0,0 +1,236 @@
+        get all copied objects so that we don't have to re-copy them
+        """
+        s3 = boto3.resource("s3")
+        existed_keys = set()
+        for _, value in PROJECT_MAP.iteritems():
+            for label in ["-open", "-controlled"]:
+                bucket_name = "gdc-" + value + label
+                bucket = s3.Bucket(bucket_name)
+                try:
+                    for file in bucket.objects.all():
+                        existed_keys.add(bucket_name + "/" + file.key)
+                except Exception:
+                    pass
+        return existed_keys
+
+    def build_source_bucket_dataset(self):
+        """
+        build source bucket dataset for lookup
+        to avoid list object operations
+        """
+        s3 = boto3.resource("s3")
+        bucket = s3.Bucket(self.bucket)
+        return set([file.key for file in bucket.objects.all()])
+
+    def exec_aws_copy(self, files):
+        """
+        Call AWS SLI to copy a chunk of  files from a bucket to another bucket.
+        Intergrity check: After each chunk copy, check the returned md5 hashes
+                            with the ones provided in manifest.
+        If not match, re-copy. Log all the success and failure cases
+
+        Args:
+            files(list): a list of files which should be copied to the same bucket
+        
+        Returns:
+            None
+        """
+
+        config_chunk_size = self.global_config.get("chunk_size", 1)
+        target_bucket = get_bucket_name(files[0], PROJECT_MAP)
+
+        index = 0
+        while index < len(files):
+            base_cmd = 'aws s3 cp s3://{} s3://{} --recursive --exclude "*"'.format(
+                self.bucket, target_bucket
+            )
+            # Some files are stored as s3://bucket/uuid
+            # cmd_for_old_files = 'aws s3 cp s3://{} s3://{}
+
+            chunk_size = min(config_chunk_size, len(files) - index)
+
+            execstr = base_cmd
+            for fi in files[index : index + chunk_size]:
+                object_name = "{}/{}".format(fi.get("id"), fi.get("filename"))
+
+                # only copy ones not exist in target bucket
+                if object_name not in self.copied_objects:
+                    if object_name in self.source_objects:
+                        execstr += ' --include "{}"'.format(object_name)
+                    elif fi.get("id") in self.source_objects:
+                        # This is an old file, just single copy
+                        cmd = "aws s3 cp s3://{}/{} s3://{}/{}".format(
+                            self.bucket, fi.get(id), target_bucket, object_name
+                        )
+                        # should wait for safety
+                        subprocess.Popen(shlex.split(cmd + " --quiet")).wait()
+
+            if execstr != base_cmd:
+                subprocess.Popen(shlex.split(execstr + " --quiet")).wait()
+
+            index = index + chunk_size
+
+        global TOTAL_PROCESSED_FILES
```

Comment:
this isn't necessary anymore right?

Reply:
just removed in caac0117d8f83af97684b250b3a50abfb033f05c